messenger/multiple_buses.rst

```diff
@@ -0,0 +1,288 @@
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <defaults>
+                   <bind key="$commandBus" type="service" id="messenger.bus.commands" />
+                   <bind key="$queryBus" type="service" id="messenger.bus.queries" />
+                   <bind key="$eventBus" type="service" id="messenger.bus.events" />
+                </defaults>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+
+        $container->bind('$commandBus', 'messenger.bus.commands');
+        $container->bind('$queryBus', 'messenger.bus.queries');
+        $container->bind('$eventBus', 'messenger.bus.events');
+
+Restrict handlers per bus & PSR-4 discovery
+-------------------------------------------
+
+By using the ``messenger.message_handler`` tag on a service, you're declaring
+it as a handler. But without more attributes, it would actually be available to
+any bus. As you don't want to dispatch a message to the wrong bus without notice,
+you can use the ``bus``  attribute with the expected bus name as value:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            App\MessageHandler\SomeCommandHandler:
+                tags: [{ name: messenger.message_handler, bus: messenger.bus.commands }]
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="App\MessageHandler\SomeCommandHandler">
+                    <tag name="messenger.message_handler" bus="messenger.bus.commands" />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+
+        $container->services()
+            ->set(App\MessageHandler\SomeCommandHandler::class)
+            ->tag('messenger.message_handler', ['bus' => 'messenger.bus.commands']);
+
+This way, the ``App\MessageHandler\SomeCommandHandler`` handler will only be
+known by the ``messenger.bus.commands`` bus.
+
+You can also leverage PSR-4 discovery to configure your handlers automatically:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        command_handlers:
+            namespace: App\MessageHandler\
```

Comment:
To follow how it's normally done, should we just use this as the key above? And then not have this `namespace` config key?